[CI] Install dill in ci

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #116215
* __->__ #116214
* #116230



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

peterbell10 committed:
[CI] Install dill in ci

[ghstack-poisoned]

peterbell10 committed:
Update on "[CI] Install dill in ci"




[ghstack-poisoned]

peterbell10 committed:
Update on "[CI] Install dill in ci"




[ghstack-poisoned]

peterbell10 committed:
Update on "[CI] Install dill in ci"




[ghstack-poisoned]

peterbell10 committed:
Update on "[CI] Install dill in ci"




[ghstack-poisoned]

peterbell10 committed:
Update on "[CI] Install dill in ci"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

peterbell10 committed:
Update on "[CI] Install dill in ci"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

peterbell10 committed:
Update on "[CI] Install dill in ci"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

peterbell10 committed:
Update on "[CI] Install dill in ci"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

peterbell10 committed:
xfail test_multiprocessing_iterdatapipe with dill on "[CI] Install dill in ci"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

peterbell10 commented on test/test_dataloader.py:
```diff
@@ -1613,6 +1599,18 @@ def test_multiprocessing_iterdatapipe(self):
+    def test_multiprocessing_iterdatapipe(self):
+        self._test_multiprocessing_iterdatapipe(with_dill=False)
+
+    @unittest.expectedFailure
```
This test fails in CI when the `HAS_DILL` code is active, so keeping the non-dill version active and xfailing the version with dill for now.

lezcano commented:
ping @malfet

malfet commented on .ci/docker/requirements-ci.txt:
```diff
@@ -25,6 +25,11 @@ coremltools==5.0b5
 #Pinned versions:
 #test that import:
 
+dill
```
Please specify the version that you want to use.

malfet reviewed (approved):
Looks good to me, but it feels like this PR solves the problem when tests are running with dill, but does not test that PyTorch will not accidentally introduce unconditional dependency on dill, because its present in all CI configs.

Can we somehow keep one of the Linux CPU configs without dill(i.e. modify one of the test configs not uninstall dill before running the tests)

peterbell10 committed:
Update on "[CI] Install dill in ci"


We have CI tests that depend on `dill` but none of the CI runners have it installed. This installs `dill` and fixes a bunch of broken tests and refactoring how dill is imported to avoid changing the default pickle behavior.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

peterbell10 committed:
Update on "[CI] Install dill in ci"


We have CI tests that depend on `dill` but none of the CI runners have it installed. This installs `dill` and fixes a bunch of broken tests and refactoring how dill is imported to avoid changing the default pickle behavior.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

peterbell10 committed:
Uninstall in docker on "[CI] Install dill in ci"


We have CI tests that depend on `dill` but none of the CI runners have it installed. This installs `dill` and fixes a bunch of broken tests and refactoring how dill is imported to avoid changing the default pickle behavior.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

peterbell10 committed:
Update on "[CI] Install dill in ci"




[ghstack-poisoned]

peterbell10 committed:
Update on "[CI] Install dill in ci"




[ghstack-poisoned]
